test/semantics/dosemantics03.f90

```diff
@@ -106,6 +123,52 @@ PROGRAM do_issue_458
     PRINT *, "plvar is: ", plvar
   END DO
 
+! SUBROUTINE DO variable
+!ERROR: DO control must be an INTEGER variable
+  DO sub = 1, 10, 3
+    PRINT *, "ivar is: ", ivar
+  END DO
+
+! FUNCTION DO variable
+!ERROR: DO control must be an INTEGER variable
+  DO ifunc = 1, 10, 3
+    PRINT *, "ivar is: ", ivar
+  END DO
+
+! POINTER to FUNCTION DO variable
+!ERROR: DO control must be an INTEGER variable
+  DO pifunc = 1, 10, 3
+    PRINT *, "ivar is: ", ivar
+  END DO
+
+! Array DO variable
+!ERROR: Must be a scalar value, but is a rank-1 array
+  DO avar = 1, 10, 3
+    PRINT *, "plvar is: ", plvar
+  END DO
+
+! Undeclared DO variable
+!ERROR: No explicit type declared for 'undeclared'
+!ERROR: DO controls should be INTEGER
```

Comment:
See previous comments. The second error is superfluous.

Reply:
Thanks, Tim.  I just pushed a fix for this.